test/framework/cluster_proxy.go

```diff
@@ -176,15 +176,17 @@ func (p *clusterProxy) GetClient() client.Client {
 	config := p.GetRESTConfig()
 
 	var c client.Client
-	err := wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
+	var retryError error
+	_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
 		var newClientErr error
 		c, newClientErr = client.New(config, client.Options{Scheme: p.scheme})
 		if newClientErr != nil {
-			return false, newClientErr
+			retryError = newClientErr
+			return false, nil //nolint:nilerr
 		}
 		return true, nil
 	})
-	Expect(err).ToNot(HaveOccurred(), "Failed to get controller-runtime client")
+	Expect(retryError).ToNot(HaveOccurred(), "Failed to get controller-runtime client")
```

Comment:
We should reset the `retryError` to `nil` at the beginning of the `PollImmediate` block or else this function will return an error if the first attempt and a subsequent attempt succeeds. May be we can refactor this to something like:
```go
var c client.Client
var newClientErr error
_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
	c, newClientErr = client.New(config, client.Options{Scheme: p.scheme})
	if newClientErr != nil {
		return false, nil //nolint:nilerr
	}
	return true, nil
})
Expect(newClientErr).ToNot(HaveOccurred(), "Failed to get controller-runtime client")
```

Reply:
Your comment actually brings up another problem, which is that we're not accounting for the first client connection attempt timing out before returning (we're throwing away the error to `wait.PollImmediate`). I'll address both cases, thanks!